src/web_app/src/pages/operation.tsx

```diff
@@ -0,0 +1,319 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-argument */
+/* eslint-disable @typescript-eslint/no-floating-promises */
+import React, {useCallback, useEffect} from "react";
+import GlobalStyles from "@mui/material/GlobalStyles";
+import CssBaseline from "@mui/material/CssBaseline";
+import {AppbarStyle} from "../components/appBar";
+import http from "../http-common";
+import {
+    LineChart, Line, XAxis, YAxis, CartesianGrid, Tooltip, Legend
+} from "recharts";
+import Container from "@mui/material/Container";
+import Typography from "@mui/material/Typography";
+import Grid from "@mui/material/Grid";
+import Button from "@mui/material/Button";
+import RoadGraph from "../components/roadGraph";
+import {useLocation} from "react-router-dom";
+import {d3Types} from "../types/d3Types";
+import {Types} from "../types/data";
+import sleep from "../config/sleep_funct";
+
+export function OperationPage() {
+    const url = useLocation().pathname;
+    const dataset_choice = url.split("/")[2];
+    const method_choice = url.split("/")[4];
+    const [
+        method_str,
+        setMethod_str
+    ] = React.useState("");
+    const [
+        data,
+        setData
+    ] = React.useState<d3Types.d3Graph>(
+        {nodes: [], links: []});
+    const [
+        data1,
+        setData1
+    ] = React.useState<[Types.Received_data]>([
+        {
+            "time"   : 0,
+            "fitness": 0
+        }
+    ]);
+    const [
+        data2,
+        setData2
+    ] = React.useState<[Types.Received_data]>([
+        {
+            "time"   : 0,
+            "fitness": 0
+        }
+    ]);
+    const [
+        data3,
+        setData3
+    ] = React.useState<[Types.Received_data]>([
+        {
+            "time"   : 0,
+            "fitness": 0
+        }
+    ]);
+    const [
+        data4,
+        setData4
+    ] = React.useState<[Types.Received_data]>([
+        {
+            "time"   : 0,
+            "fitness": 0
+        }
+    ]);
+    const [
+        start,
+        setStart
+    ] = React.useState<boolean>(false);
+    const [
+        enableButton,
+        setEnableButton
+    ] = React.useState(false);
+    useEffect(() => {
+        (
+            async() => {
+                await http.get(`get_first_sol?${dataset_choice}`)
+                    .then(response_f => {
```

Comment:
Strange! I am not sure that this information is true. I need this to check that everything is working well or to save the response(data) and use it.

Reply:
I did a little more research and can summarize what I found in a few points : 
- it will work (async/await is just an alternative way to work with promises)
- it can be confusing and confusion can lead to bugs. 

This is therefore not a big problem. As of the confusion, I do not know others feel about it. So as said earlier this is not “recommended” but not a high priority thing I guess.